partner_firstname/models/firstname_mixin.py

```diff
@@ -0,0 +1,171 @@
+            name = name.decode("utf-8")
+
+        try:
+            name = " ".join(name.split()) if name else name
+        except UnicodeDecodeError:
+            # with users coming from LDAP, name can be a str encoded as utf-8
+            # this happens with ActiveDirectory for instance, and in that case
+            # we get a UnicodeDecodeError during the automatic ASCII -> Unicode
+            # conversion that Python does for us.
+            # In that case we need to manually decode the string to get a
+            # proper unicode string.
+            name = " ".join(name.decode("utf-8").split()) if name else name
+
+        if comma:
+            name = name.replace(" ,", ",")
+            name = name.replace(", ", ",")
+        return name
+
+    @api.model
+    def _get_computed_name(self, lastname, firstname):
+        """Compute the 'name' field according to splitted data.
+        You can override this method to change the order of lastname and
+        firstname the computed name"""
+        order = self._get_names_order()
+        if order == "last_first_comma":
+            return ", ".join(p for p in (lastname, firstname) if p)
+        elif order == "first_last":
+            return " ".join(p for p in (firstname, lastname) if p)
+        else:
+            return " ".join(p for p in (lastname, firstname) if p)
+
+    @api.model
+    def _get_inverse_name(self, name, is_company=False):
+        """Compute the inverted name.
+
+        - If the partner is a company, save it in the lastname.
+        - Otherwise, make a guess.
+
+        This method can be easily overriden by other submodules.
+        You can also override this method to change the order of name's
+        attributes
+
+        When this method is called, :attr:`~.name` already has unified and
+        trimmed whitespace.
+        """
+        # Company name goes to the lastname
+        if is_company or not name:
+            parts = [name or False, False]
+        # Guess name splitting
+        else:
+            order = self._get_names_order()
+            # Remove redundant spaces
+            name = self._get_whitespace_cleaned_name(
+                name, comma=(order == "last_first_comma")
+            )
+            parts = name.split("," if order == "last_first_comma" else " ", 1)
+            if len(parts) > 1:
+                if order == "first_last":
+                    parts = [" ".join(parts[1:]), parts[0]]
+                else:
+                    parts = [parts[0], " ".join(parts[1:])]
+            else:
+                while len(parts) < 2:
+                    parts.append(False)
+        return {"lastname": parts[0], "firstname": parts[1]}
+
+    @api.model
+    def name_fields_in_vals(self, vals):
+        """Method to check if any name fields are in `vals`."""
+        return vals.get("firstname") or vals.get("lastname")
+
+    @api.constrains("name", "firstname", "lastname")
+    def _check_name(self):
```

Comment:
Here you need to adapt again the method name otherwise it will be not triggered. 
Like it is described here. https://github.com/OCA/partner-contact/pull/2174#discussion_r2602825234